opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/async/TraceAsyncAspect.java

```diff
@@ -0,0 +1,70 @@
+package io.opentracing.contrib.spring.cloud.async;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.NoopActiveSpanSource;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.jms.common.TracingMessageListener;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.ReflectionUtils;
+
+import javax.jms.Message;
+import javax.jms.MessageListener;
+import java.lang.reflect.Method;
+
+/**
+ * @author kamesh
+ */
+@Aspect
+public class TraceAsyncAspect {
+
+    private static final String ASYNC_COMPONENT = "async";
+    public static final String TAG_LC = "localcomponent";
+    public static final String TAG_CLASS = "class";
+    public static final String TAG_METHOD = "method";
+
+    Tracer tracer;
+    protected final ActiveSpan.Continuation continuation;
```

Comment:
Is this aspect singleton or instance per-invocation?

Reply:
this is per invocation? why?